jupyterlab_bigquery/src/utils/widget_manager.ts

```diff
@@ -9,29 +9,44 @@ import { ReduxReactWidget } from './ReduxReactWidget';
  * A class that manages dataset widget instances in the Main area
  */
 export class WidgetManager {
+  private static instance: WidgetManager = undefined;
   private widgets: { [id: string]: Widget } = {};
   private reduxWidgets: { [id: string]: ReduxReactWidget } = {};
   private store: EnhancedStore;
 
-  constructor(private app: JupyterFrontEnd) {
+  private constructor(private app: JupyterFrontEnd) {
     this.store = configureStore({ reducer: rootReducer });
 
     return this;
   }
 
+  static getInstance(app?: JupyterFrontEnd): WidgetManager {
```

Comment:
I think this would be better to split into `initInstance(app: JupyterFrontEnd)` and `getInstance(): WidgetManager` rather than combining into one concept.

Reply:
@MRzNone here are the changes to the widget manager